pennylane/_grad.py

```diff
@@ -21,55 +21,101 @@
 
+    .. note::
+
+        When used with :func:`~.qjit`, this function only supports the Catalyst compiler.
+        Please see :func:`catalyst.grad` for more details.
+
+        Please see the Catalyst :doc:`quickstart guide <catalyst:dev/quick_start>`,
+        as well as the :doc:`sharp bits and debugging tips <catalyst:dev/sharp_bits>`
+        page for an overview of the differences between Catalyst and PennyLane.
+
     Args:
         func (function): a plain QNode, or a Python function that contains
             a combination of quantum and classical nodes
+
         argnum (int, list(int), None): Which argument(s) to take the gradient
             with respect to. By default, the arguments themselves are used
             to determine differentiability, by examining the ``requires_grad``
             property.
 
+        method (str): Specifies the gradient method when used with
+                      the :func:`~.qjit` decorator. Outside of the
+                      :func:`~.qjit`, this keyword argument has no effect.
+                      and should not be set.
+                      In just-in-time (JIT) mode, this can be any of ``["auto", "fd"]``, where:
+
+                      - ``"auto"`` represents deferring the quantum differentiation to the method
+                        specified by the QNode, while the classical computation is differentiated
+                        using traditional auto-diff. Catalyst supports ``"parameter-shift"`` and
+                        ``"adjoint"`` on internal QNodes. Notably, QNodes with
+                        ``diff_method="finite-diff"`` are not supported with ``"auto"``.
+
+                      - ``"fd"`` represents first-order finite-differences for the entire hybrid
+                        function.
+
+        step_size (float): The step-size value for the finite-difference (``"fd"``) method within
+                      :func:`~.qjit` decorated functions. This value has
+                      no effect in non-compiled functions.
+
     Returns:
         function: The function that returns the gradient of the input
         function with respect to the differentiable arguments, or, if specified,
         the arguments in ``argnum``.
     """
 
-    def __init__(self, fun, argnum=None):
+    def __new__(cls, func, argnum=None, method=None, step_size=None):
+        """Patch to the proper grad function"""
+
+        if active_jit := compiler.active_compiler():
+            available_eps = compiler.AvailableCompilers.names_entrypoints
+            ops_loader = available_eps[active_jit]["ops"].load()
+            return ops_loader.grad(func, method=method, h=step_size, argnum=argnum)
+
+        if method or step_size:
+            raise ValueError(
+                f"Invalid values for 'method={method}' and 'step_size={step_size}' in interpreted mode"
+            )
+
+        return super().__new__(cls)
+
+    def __init__(self, func, argnum=None):
```

Comment:
yep, but it's now aligned with the docstring of the `grad` op :smile: But, this is not essential for this work and we can revert this if this will break anything.

Reply:
omg the docstring arg name has been wrong this whole time! in that case, I'd keep your change to `func` here 👍